index.js

```diff
@@ -134,6 +145,94 @@ function ScreenshotReporter(options) {
  	}
 }
 
+var currentSuite, currentSpec, startTime;
+
+ScreenshotReporter.prototype.jasmineStarted = function () {
+    //console.log("##test[progressStart 'Running Jasmine Tests']");
+    var self = this;
+
+    afterEach (function(next){
+        /** Function: specDone - THIS WAS in specDone, but that was causing async issues
+         * Called by Jasmine when a test spec is done. It triggers the
+         * whole screenshot capture process and stores any relevant information.
+         *
+         * Parameters:
+         *     (Object) spec - The test spec to report.
+         */
+
+        var results = spec = currentSpec;
+        if(!self.takeScreenShotsForSkippedSpecs && results.skipped) {
+            return;
+        }
+
+        spec.totalTime = new Date().getTime() - startTime;
+        browser.takeScreenshot().then(function (png) {
+            browser.getCapabilities().then(function (capabilities) {
+                var descriptions = util.gatherDescriptions(
+                        currentSuite
+                        , [spec.description]
+                    )
+
+                    , baseName = self.pathBuilder(
+                        spec
+                        , descriptions
+                        , results
+                        , capabilities
+                    )
+                    , metaData = self.metaDataBuilder(
+                        spec
+                        , descriptions
+                        , results
+                        , capabilities
+                    )
+
+                    , screenShotFile = baseName + '.png'
+                    , metaFile = baseName + '.json'
+                    , screenShotPath = path.join(self.baseDirectory, screenShotFile)
+                    , metaDataPath = path.join(self.baseDirectory, metaFile)
+
+                // pathBuilder can return a subfoldered path too. So extract the
+                // directory path without the baseName
+                    , directory = path.dirname(screenShotPath);
+
+                metaData.screenShotFile = screenShotFile;
+                mkdirp(directory, function(err) {
+                    if(err) {
+                        throw new Error('Could not create directory ' + directory);
+                    } else {
+                        util.addMetaData(metaData, metaDataPath, descriptions, self.finalOptions);
+
+                        if(!(self.takeScreenShotsOnlyForFailedSpecs && results.status === 'passed')) {
+                            util.storeScreenShot(png, screenShotPath);
+                        }
+                        util.storeMetaData(metaData, metaDataPath);
+                    }
+                });
+                next();
```

Comment:
```mkdirp``` is async.  You will need to move it in the callback for that function call to ensure you don't move on before the data is written to disk.